crates/ide_completion/src/render.rs

```diff
@@ -313,17 +306,44 @@ impl<'a> Render<'a> {
     }
 }
 
-fn compute_relevance(ctx: &RenderContext, ty: &Type, name: &str) -> CompletionRelevance {
-    let mut res = CompletionRelevance::default();
+fn compute_exact_type_match(ctx: &CompletionContext, completion_ty: &hir::Type) -> bool {
+    if let Some(expected_type) = ctx.expected_type.as_ref() {
+        // We don't ever consider unit type to be an exact type match, since
+        // nearly always this is not meaningful to the user.
+        completion_ty == expected_type && !expected_type.is_unit()
+    } else {
+        false
+    }
```

Comment:
That's not really important, but we try to avoid if let else (there's :bulb: ): https://github.com/rust-analyzer/rust-analyzer/pull/8046